Libraries/Blob/Blob.js

```diff
@@ -60,61 +53,39 @@ import type { BlobProps } from 'BlobTypes';
  * Reference: https://developer.mozilla.org/en-US/docs/Web/API/Blob
  */
 class Blob {
-  /**
-   * Size of the data contained in the Blob object, in bytes.
-   */
-  size: number;
-  /*
-   * String indicating the MIME type of the data contained in the Blob.
-   * If the type is unknown, this string is empty.
-   */
-  type: string;
-
-  /*
-   * Unique id to identify the blob on native side (non-standard)
-   */
-  blobId: string;
-  /*
-   * Offset to indicate part of blob, used when sliced (non-standard)
-   */
-  offset: number;
-
-  /**
-   * Construct blob instance from blob data from native.
-   * Used internally by modules like XHR, WebSocket, etc.
-   */
-  static create(props: BlobProps): Blob {
-    return Object.assign(Object.create(Blob.prototype), props);
-  }
+  _data: ?BlobData;
 
   /**
    * Constructor for JS consumers.
    * Currently we only support creating Blobs from other Blobs.
    * Reference: https://developer.mozilla.org/en-US/docs/Web/API/Blob/Blob
    */
-  constructor(parts: Array<Blob>, options: any) {
-    const blobId = uuid();
-    let size = 0;
-    parts.forEach((part) => {
-      invariant(part instanceof Blob, 'Can currently only create a Blob from other Blobs');
-      size += part.size;
-    });
-    BlobModule.createFromParts(parts, blobId);
-    return Blob.create({
-      blobId,
-      offset: 0,
-      size,
-    });
+  constructor(parts: Array<Blob | string> = [], options?: BlobOptions) {
+    const BlobManager = require('BlobManager');
+    this.data = BlobManager.createFromParts(parts, options).data;
   }
 
   /*
    * This method is used to create a new Blob object containing
    * the data in the specified range of bytes of the source Blob.
    * Reference: https://developer.mozilla.org/en-US/docs/Web/API/Blob/slice
    */
+  set data(data: ?BlobData) {
```

Comment:
I don't see any reason for making `data` writeable. Why not make `Blob` write to `_data` directly and replace the `data` getter with a `getData` method?

Also, these methods cause the docblock for `slice` to be misplaced. Please fix.

Reply:
IMO doesn't make sense to expose both `_data` and `data` to outside the module. It's more consistent use the same property name to read and write.